src/pages/City/CityPage.tsx

```diff
@@ -13,21 +14,13 @@ type Params = {
 const CityPage = () => {
   const { city } = useParams() as Params;
   const { data, isLoading } = Weather.useLocation(city);
-  const { state }: NavigateOptions = useLocation();
-
-  const renderCityName = () => {
-    if (isLoading) {
-      return <CityNameLoader />;
-    }
-    if (data) {
-      return <CityName>{state?.cityName || data.location.name}</CityName>;
-    }
-  };
 
   return (
     <PageContentContainer data-testid={CityPageTestIds.Container}>
-      {renderCityName()}
+      <CityNameWidget loading={isLoading} cityName={data !== undefined ? data.location.name : undefined} />
       <WidgetWrapper data={data} isLoading={isLoading} data-testid={CityPageTestIds.Widgets} />
+      <NearestCitiesTitle>nearest</NearestCitiesTitle>
```

Comment:
**suggestion** move this to the weather widget 

this title is part of the widget and it should be in widget body